wgpu-core/src/command/render.rs

```diff
@@ -1262,6 +1275,29 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                         raw.draw_indexed_indirect(&buffer.raw, offset, 1, 0);
                     }
                 }
+                RenderCommand::PushDebugGroup {
+                    color,
+                    len,
+                    phantom_marker,
+                } => unsafe {
+                    let (new_peeker, label) =
+                        { phantom_marker.decode_unaligned(peeker, len, raw_data_end) };
+                    peeker = new_peeker;
+
+                    raw.begin_debug_marker(std::str::from_utf8(label).unwrap(), color)
+                },
+                RenderCommand::PopDebugGroup {} => unsafe { raw.end_debug_marker() },
```

Comment:
can we add a single number (to the `State` structure) tracking how deep we are in the push/pop groups? We don't want to cause UB in the driver by popping a marker that wasn't pushed. Generally, the data we are decoding here is *not* trusted.